### PR TITLE
Implement rayon

### PR DESCRIPTION
**Pull request by ahqsoftwares**
This PR adds rayon support alongside crossbeam

### CODE REVIEW

File: .vscode/settings.json
Comment (Blakeinstein) [inline .vscode/settings.json:1]: Why was this formatted this way? Was wondering if we should drop this change.

File: Cargo.toml [hunk 1]
```diff
@@ -9,23 +9,29 @@ documentation = "https://docs.rs/fuse-rust"
...
-members = [
-  "examples/search-bar"
-]
+members = ["examples/search-bar"]
 
 [[example]]
 name = "chunk-search"
-required-features = ["async"]
\ No newline at end of file
+required-features = ["async"]
```
Comment (Blakeinstein) [inline Cargo.toml:37]: Think you can also create another example using rayon instead?

**Review by Blakeinstein — COMMENTED** [review]
Overall looks good, will test the code a bit later. Can you also include tests/examples as well for the rayon flag. Also was wondering if we should better naming conventions for the feature flags.

**Comment by Blakeinstein** [comment]
Please run `cargo clippy` and `cargo fmt`!

File: Cargo.toml [hunk 2]
```diff
@@ -9,23 +9,29 @@ documentation = "https://docs.rs/fuse-rust"
...
+]
 description = """
 Fuse is a super lightweight library which provides a simple way to do fuzzy searching.
 Fuse-Rust is a port of Fuse-Swift, written purely in rust.
 """
 
 [dependencies]
-crossbeam-utils = { version = "0.8.14", optional = true }
+crossbeam-utils = { version = "0.8", optional = true }
+rayon = { version = "1", optional = true }
```
Comment (Blakeinstein) [inline Cargo.toml:26]: Also why are we using version 1? rayon is at `1.10`

Comment (ahqsoftwares) [reply Cargo.toml:26]: It is to integrate it better with other crates. 1 is 1.x according to semver

File: Cargo.toml (hunk 1 above)
Comment (ahqsoftwares) [reply Cargo.toml:37]: Done 🎉

File: Cargo.toml [hunk 3]
```diff
@@ -9,23 +9,30 @@ documentation = "https://docs.rs/fuse-rust"
...
-members = [
-  "examples/search-bar"
-]
+members = ["examples/search-bar"]
 
 [[example]]
 name = "chunk-search"
-required-features = ["async"]
\ No newline at end of file
+required-features = ["async"]
```
Comment (Blakeinstein) [inline Cargo.toml:38]: We didnt add a required-features flag for `chunk-search-rayon`?

Comment (ahqsoftwares) [reply Cargo.toml:38]: yeah...

**Comment by ahqsoftwares** [comment]
CI!!!

**Pull request merged**